packages/xo-server-sdn-controller/src/private-network/private-network.js

```diff
@@ -125,18 +127,13 @@ export class PrivateNetwork {
   async electNewCenter() {
     delete this.center
 
-    // TODO: make it random
-    const hosts = this._getHosts()
-    for (const host of hosts) {
-      const pif = host.$PIFs.find(
-        _ => _.network === this.networks[host.$pool.uuid].$ref
+    if (this._preferredCenter !== undefined) {
+      this._preferredCenter = await this._preferredCenter.$xapi.barrier(
```

Comment:
Why is it necessary?

Why not for all the hosts?

Reply:
Because I only store in my object the preferred network, i need to update it to see if at the time the code is called it can be center.

The other host are get from the xapi so should be up to date.